library/VTKExtensions/Rendering/vtkF3DRenderer.cxx

```diff
@@ -1074,8 +1060,32 @@ int vtkF3DRenderer::UpdateLights()
   {
     vtkNew<vtkLightKit> lightKit;
     lightKit->AddLightsToRenderer(this);
-    return this->GetLights()->GetNumberOfItems();
+    lightCount = this->GetLights()->GetNumberOfItems();
```

Comment:
needs `this->LightIntensitiesConfigured = false` here maybe?

Reply:
also the doc for `vtkRenderer.UpdateLights()` says "This method will return the actual number of lights that were on", does that mean that there could be some "not on" lights that don't get counted? in which case the `lightCount == 0` test for this block would be better checked on `this->GetLights()->GetNumberOfItems()` directly?
sidenote:  no `isEmpty()` method in `vtkCollection`? :(

Reply:
> needs this->LightIntensitiesConfigured = false here maybe?

Indeed needed.

Reply:
> does that mean that there could be some "not on" lights that don't get counted?

Indeed, that is correct. However the only case where it matters would be if somehow a user was able to import a file with turned off lights.

In that case, I'd rather add lights to the scene instead of having a unlighted scene. I'll add a comment.

Reply:
> no isEmpty() method in vtkCollection? :(

tbf, vtkCollection should be reimplemented fully to be just a wrapper of std::vector. This class is coming from the time were VTK did not support C++11 at all.